src/main/java/com/dominik/todolist/controller/AuthController.java

```diff
@@ -0,0 +1,70 @@
+package com.dominik.todolist.controller;
+
+import com.dominik.todolist.dto.AuthResponse;
+import com.dominik.todolist.dto.LoginRequest;
+import com.dominik.todolist.dto.RegisterRequest;
+import com.dominik.todolist.exception.EmailAlreadyExistsException;
+import com.dominik.todolist.model.AppUser;
+import com.dominik.todolist.security.JwtUtil;
+import com.dominik.todolist.service.UserService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    private final UserService userService;
+    private final AuthenticationManager authenticationManager;
+    private final JwtUtil jwtUtil;
+
+    @Autowired
```

Comment:
NIT:
I think that currently Spring is autowiring automatically constructors in @Components